src/main/services/catalog.ts

```diff
@@ -251,11 +253,12 @@ export class CatalogService {
                 mainWindow = window;
             }
         }
-
+        store.dispatch(exportDialogOpen());
```

Comment:
Rather than having usage-specific actions for each call to `dialog.showOpenDialog()` (and specific store properties, like `export`, `publication info`, etc.), I think it would be better to have a single "dialog open/close" action type, with a parameter indicating its usage. This generic design would allow reuse of a single construct in different usage contexts (wherever a modal dialog is open/closed). In fact this could/should be generalized even further to any dialog/popup menu/etc. which needs to return keyboard focus to a calling site.

Reply:
I am proposing to close this PR, which unfortunately has several key conflicts to resolve due to recent codebase evolutions (notably: strongly-typed Redux actions).